src/distrs/wilcox.jl

```diff
@@ -0,0 +1,149 @@
+#=
+The wilcox distribution is equivalent to the problem
+of finding the number of subsets with size nx
+of {1,2,...,nx,nx+1,...,nx+ny} summing to (U+sum(1:nx))
+relative to the total number of subsets with size nx.
+The empty subset is defined to sum to zero.
+This can be calculated using the recursion:
+either nx+ny is in the subset in which case we need to calculate
+the number of subsets with size nx-1 of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U+sum(1:nx)-nx-ny),
+or nx is not in the subset in which case we need to calculate
+the number of subsets with size nx of {1,2,...,nx,nx+1,...,nx+ny-1}
+summing to (U+sum(1:nx)),
+This can be calculated bottom up using dynamic programming.
+
+The (i,j)'th element of DP in the k'th outer loop iteration represents:
+the number of subsets with size k of {1,2,...,k,k+1,...,k+j-1} summing to i+sum(1:k)-1.
+ =#
+
+@inline function wilcoxDP(nx, ny, U)
+    DP = zeros(Int, U + 1, ny + 1)
+    for j = 1:(ny+1)
+        DP[1, j] = 1
+    end
+    for k = 1:nx
+        for j = 2:(ny+1)
+            for i = (U+1):-1:2
+                DP[i, j] = DP[i, j-1]
+                if i - j >= 0
+                    DP[i, j] += DP[i-j+1, j]
+                end
+            end
+        end
+    end
+    return DP
+end
+
+function wilcoxpdf(nx::Int, ny::Int, U::Float64)
+    return isinteger(U) ? wilcoxpdf(nx, ny, Int(W)) : 0.0
+end
+function wilcoxpdf(nx::Int, ny::Int, U::Int)
+    if U < 0
+        return 0.0
+    end
+    max_U = nx * ny
+    U2 = max_U - U
+    if U2 < U
+        return wilcoxpdf(nx, ny, U2)
+    end
+    DP = wilcoxDP(nx, ny, U)
+    DP[U+1, ny+1] / binomial(nx + ny, nx)
+end
+
+function wilcoxlogpdf(nx::Int, ny::Int, U::Union{Float64,Int})
+    return log(wilcoxpdf(nx, ny, U))
+end
+
+function wilcoxcdf(nx::Int, ny::Int, U::Float64)
+    wilcoxcdf(nx, ny, round(Int, U, RoundNearestTiesUp))
+end
+function wilcoxcdf(nx::Int, ny::Int, U::Int)
+    if U < 0
+        return 0.0
+    end
+    max_U = nx * ny
+    U2 = max_U - U - 1
+    if U2 < U
+        return 1.0 - wilcoxcdf(nx, ny, U2)
+    end
+    DP = wilcoxDP(nx, ny, U)
+    sum(@view(DP[1:(U+1), ny+1]) ./ binomial(nx + ny, nx))
```

Comment:
To avoid an intermediate array:
```suggestion
    sum(@view(DP[1:(U+1), ny+1])) / binomial(nx + ny, nx)
```

Reply:
Included it, but doesn't seem to make a difference to allocations on my machine.